lib/mvi/responses/base.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+module MVI
+  module Responses
+    class Base
+      attr_accessor :code, :query, :original_response
+
+      RESPONSE_CODES = {
+        success: 'AA',
+        failure: 'AE',
+        invalid_request: 'AR'
+      }.freeze
+
+      class << self
+        attr_accessor :endpoint
+      end
+
+      def self.mvi_endpoint(endpoint)
```

Comment:
Agreed with @omgitsbillryan This creates a DSL for retrieving the right section of data from the response, but it is a bit more confusing that it needs to be. Could this just be a property of the class and the subclasses can override it?